### PR TITLE
Adds Requisition to Marine Headsets

### PR DESCRIPTION
**Pull request by WinterDarkraven**
## About The Pull Request
Adds requisitions to marine headsets
## Why It's Good For The Game

RO is a stressful job in a video game which leads to a lot of confusion and miscommunication. This helps streamline the process. 

Instead of RO's having to keep mental, or sometimes physical, lists of what each marine wants, a marine can now use the ALREADY existing list feature that gets ignored once the game begins -- the request menu!

---

**The radio interface cannot be used inside any level of caving, even shallow caves. Get a radio backpack for that.**

---

The goal of this PR is to make RO more enjoyable, as a lot of players do not like the stress that comes with it, and will make the job more friendly to new players that do not know where everything is. I feel like this will make the game more enjoyable for everyone without stepping on too many toes, as this is giving marines nothing they already couldn't do.

**This does NOT give marines a free beacon.** Marines and the RO will still have to communicate about the arrival of their order, and work with their SL or FC for their equipment -- or even request a **radio backpack to use the req menu inside caves** and for speedier deliveries! RO will still have to fill boxes of ammo, though.

---

Since the antenna is now no longer a free beacon and doesn't seem to be coming back, this change is even better, as it will still give the radio backpack a use as a beacon and won't encourage solo-play. 

---

~~Now, I know what you're thinking, "Can't Valhalla use this???!!??!"~~

~~And the answer is... yes! They can! They always could! The radio backpack works in valh! I'm new to coding and don't know how to fix this, so that might need fixed first. Don't tell anyone I said that, though.~~

# The valh bug is fixed. Both the previous radio backpack bug and the new radio req bug are fixed. Both no longer work at valh

---

https://github.com/tgstation/TerraGov-Marine-Corps/assets/70779744/84abe098-b5b0-4736-af82-6299c5e9bed7

**Please note that in the video there's an "ammo" tab. It is from an old version of this PR. There is no ammo tab in the current version.**

---
Still new to coding, so there might be some problems, but I'll work them out the best I can.
## Changelog
:cl:
add: TGMC has installed a new receiver into all TGMC radios to let them communicate their needs to requisitions more efficiently. Due to hardware limitations, the size of the receiver cannot transmit underground, buy a TGMC Radio Backpack today for cleaner transmissions underground!
/:cl:

### CODE REVIEW

**Comment by WinterDarkraven** [comment]
Oh, this will probably work on crash too, but the radio backpack also does. So... ummm

**Comment by FosterGrime-Heymynameslouie** [comment]
blessed be speedgmerhg

**Comment by WinterDarkraven** [comment]
> Not da powercreep being passed as qol.

There might be some confusion. This isn't giving every marine a beacon or anything, it only lets the marine access the request menu. I also wanted to let the RO edit requests so they can lower the amount of things they're buying for someone, but I don't know how to do that yet.

I'll probably make a lot more small edits to req if this goes through. Had an idea for a priority tag people can put on their requests, but again... am... stoppopid.... plus everyone would put their requests to critical importance.

**Comment by SzkieletorNutric** [comment]
Holy powercrêpe

**Comment by Ragecrabbo** [comment]
Just add req console to FOB or something

**Comment by WinterDarkraven** [comment]
> Holy powercrêpe

I really don't see how this is giving marines anything they can't already do, it just off-loads the work off of the RO.

**Comment by usnpeepoo** [comment]
If this change allows a marine to order req stuff while in caves than it's seriously just powercreep

**Comment by WinterDarkraven** [comment]
> If this change allows a marine to order req stuff while in caves than it's seriously just powercreep

Fair, didn't think of that. I'll see if I can snatch some of the cave radio suppression thing and slap it onto the req display.

**Comment by WinterDarkraven** [comment]
Updated to make the radios not work in any level of caves, at all. If anyone else has a real concern, please let me know! I'm actively learning and making changes!

**Comment by breadhunt** [comment]
Req probably needs this

**Comment by WinterDarkraven** [comment]
Just noticed there was an old grammar mistake with the radio backpack, which got carried over to the radios. Both are fixed, plus more player feedback was added to the caves, so players don't think the req menu is just broken

image

**Comment by Kuro020** [comment]
Gonna have to think about this one

**Comment by bowlofcereal** [comment]
this should probably get the balance tag because it will indefinitely effect the game balance

**Comment by WinterDarkraven** [comment]
> this should probably get the balance tag because it will indefinitely effect the game balance

I do plan on making req guns and equipment less annoying for xeno players. However, req in its current state is a bit of a mess. Points should hopefully keep the marine-side in check for awhile, but marines get a lot of points simply for existing. I'll probably go after the 5 orbit meta if it's okay to do that later.

**Comment by Derrick2Electricboogaloo** [comment]
> > Oh, this will probably work on crash too, but the radio backpack also does. So... ummm
> 
> Yeah Radio pack is pretty good, also it's not really expensive to get either from req

This, use radio pack if you want the Req QOL otherwise, why even have radiopack anymore if this will just invalidate it completely. What would be the point? Other than taking up a backpack slot to drop a beacon instead of your helmet mod.

**Comment by WinterDarkraven** [comment]
> > > Oh, this will probably work on crash too, but the radio backpack also does. So... ummm
> > 
> > 
> > Yeah Radio pack is pretty good, also it's not really expensive to get either from req
> 
> This, use radio pack if you want the Req QOL otherwise, why even have radiopack anymore if this will just invalidate it completely. What would be the point? Other than taking up a backpack slot to drop a beacon instead of your helmet mod.

The Radio Backpack function is now as a beacon and to access req inside caves, the standard radio interface cannot access req while in any level of caves, like disks for example. They are still good, especially for plat mining, which you cannot do with a standard radio, even if you had a beacon.

**Comment by WinterDarkraven** [comment]
> > > > Oh, this will probably work on crash too, but the radio backpack also does. So... ummm
> > > 
> > > 
> > > Yeah Radio pack is pretty good, also it's not really expensive to get either from req
> > 
> > 
> > This, use radio pack if you want the Req QOL otherwise, why even have radiopack anymore if this will just invalidate it completely. What would be the point? Other than taking up a backpack slot to drop a beacon instead of your helmet mod.
> 
> The Radio Backpack function is now as a beacon and to access req inside caves, the standard radio interface cannot access req while in any level of caves, like disks for example. They are still good, especially for plat mining, which you cannot do with a standard radio, even if you had a beacon.

This is listed in the main PR. Again, I'm trying to not step on as many toes as possible. If you have any suggestions, I'll happily include them.

**Comment by SimplySmiley123** [comment]
> Req itself is stupid and is need of a nerf, making it more streamlined and balancing it out seems like a good option.

In need of a nerf when it already is a pain in the ass to work with.
For what, to make no one interact with it at all?

**Comment by TellyLsc** [comment]
> > Req itself is stupid and is need of a nerf, making it more streamlined and balancing it out seems like a good option.
> 
> In need of a nerf when it already is a pain in the ass to work with. For what, to make no one interact with it at all?

Yes, because reading an everscrolling text box for a brown text that says "RO, SEND SG-85 AMMO BINS" is interaction instead of checking your req console and seeing that there's an "SG-85 ammo bin x3 order by whatshisname SG, reason: I need more boolits"

**Comment by Pariah919** [comment]
>The Radio Backpack function is now as a beacon and to access req inside caves, the standard radio interface cannot access req while in any level of caves.


What? That's bad. Someone should change that. Caves shouldn't have any direct ship to ground form of logistics.

**Comment by WinterDarkraven** [comment]
> > > The Radio Backpack function is now as a beacon and to access req inside caves, the standard radio interface cannot access req while in any level of caves.
> > 
> > 
> > What? That's bad. Someone should change that. Caves shouldn't have any direct ship to ground form of logistics.
> 
> I could add the req interface code to the regular radio backpack, but right now people use the radio backpack for plat mining, but I don't think anyone wants the radio back nerfed...
> 
> But that could be another PR.

**Comment by WinterDarkraven** [comment]
> > The Radio Backpack function is now as a beacon and to access req inside caves, the standard radio interface cannot access req while in any level of caves.
> 
> What? That's bad. Someone should change that. Caves shouldn't have any direct ship to ground form of logistics.

I actually really like this idea. The req menu isn't the problem -- beacons are. I wouldn't mind making beacons not work in caves, it would make plat mining more risky and an investment, since you'd have to bring all the mats you need TO the plat miner and not just building up into a super fortress while making points.

Could remove the req interface from the radio backpack as legacy code and make the backpack only a beacon, which is still insanely good, since beacons are so damn powerful antenna had to be removed. 

Just a thought, throwing ideas out there.

**Comment by chizzy376** [comment]
At what point does qol step on the boundaries of balance?
You could argue that every qol PR is affecting balance, as it makes the life of someone easier and thus lets them focus on more important things, though that would be a very petty take that _I'm sure nobody on this repo would have ;)_ 
As for this PR, let's discuss what this streamlines: Ordering from req. Not actually getting your items delivered, but being able to access the req menu at will.
**What does this do?** Of course, it makes ordering from req without having a radiopack yourself a 100 times easier if it's a bigboy reqmenu-only order.
But let's discuss the other things this does: It lets marines **see** the current points of req as well as any pending/accepted/denied orders. This isn't a huge deal like _some_ might want you to think, but it's good to have knowledge of to track your ~~DHL shipment of your~~ B18



TL:DR; QoL good, no more yelling at each other for ordering (tho yelling for sending/receiving stays :p )

**Comment by Aporhtonoma** [comment]
im a fan of changes towards decluttering the req chat.Nothing is more depressing as req man than to get 20 diffrent orders.Most of them impossible. At least 15 of said orders will be on the orders tab now.

**Comment by Autisem** [comment]
As for the valhala bug, you could propably make the req interface inaccessible if your valhala mob(check the job associated with the mob,should be fallen something).

Something to be said about the fact marines can now see the amount of points you have, but I dont care that much about that.

Also to fix the free stuff problem, you could put them in wooden crates instead of the steel crates, those are compatible with the launch pad. OR make them not map to ANY actuel supply item. simply having them as a record you can request but it wont deliver anything

**Comment by WinterDarkraven** [comment]
> As for the valhala bug, you could propably make the req interface inaccessible if your valhala mob(check the job associated with the mob,should be fallen something).
> 
> Something to be said about the fact marines can now see the amount of points you have, but I dont care that much about that.
> 
> Also to fix the free stuff problem, you could put them in wooden crates instead of the steel crates, those are compatible with the launch pad. OR make them not map to ANY actuel supply item. simply having them as a record you can request but it wont deliver anything

I'll see if I can find a way to add a job check to both the radio backpack and req radio, thanks.

The ammo feature was just bad design in hindsight. It probably won't be coming back.

**Comment by Kuro020** [comment]
I forgot about this
